lib/ansible/modules/cloud/amazon/iam_server_certificate_info.py

```diff
@@ -36,136 +42,166 @@
-    type: str
-    sample: "2015-04-25T00:36:40+00:00"
+    type: complex
+    contains:
+        certificate_body:
+            description: The asn1der encoded PEM string
+            returned: always
+            type: str
+            sample: "-----BEGIN CERTIFICATE-----\nbunch of random data\n-----END CERTIFICATE-----"
+        certificate_chain:
+            description: The asn1der encoded PEM string
+            returned: always
+            type: str
+            sample: "-----BEGIN CERTIFICATE-----\nbunch of random data\n-----END CERTIFICATE-----"
+        server_certificate_metadata:
+            description: Returns an array of complex objects as described below.
+            returned: always
+            type: complex
+            contains:
+                server_certificate_id:
+                    description: The 21 character certificate id
+                    returned: always
+                    type: str
+                    sample: "ADWAJXWTZAXIPIMQHMJPO"
+                server_certificate_name:
+                    description: The name of the server certificate
+                    returned: always
+                    type: str
+                    sample: "server-cert-name"
+                arn:
+                    description: The Amazon resource name of the server certificate
+                    returned: always
+                    type: str
+                    sample: "arn:aws:iam::911277865346:server-certificate/server-cert-name"
+                path:
+                    description: The path of the server certificate
+                    returned: always
+                    type: str
+                    sample: "/"
+                expiration:
+                    description: The date and time this server certificate will expire, in ISO 8601 format.
+                    returned: always
+                    type: str
+                    sample: "2017-06-15T12:00:00+00:00"
+                upload_date:
+                    description: The date and time this server certificate was uploaded, in ISO 8601 format.
+                    returned: always
+                    type: str
+                    sample: "2015-04-25T00:36:40+00:00"
 '''
 
+from ansible.module_utils.aws.core import AnsibleAWSModule
+from ansible.module_utils.ec2 import AWSRetry, camel_dict_to_snake_dict
 
+# Non-ansible imports
 try:
-    import boto3
-    import botocore.exceptions
-    HAS_BOTO3 = True
+    import botocore
 except ImportError:
-    HAS_BOTO3 = False
+    pass  # caught by AnsibleAWSModule
 
-from ansible.module_utils.basic import AnsibleModule
-from ansible.module_utils.ec2 import boto3_conn, ec2_argument_spec, get_aws_connection_info
 
+@AWSRetry.backoff(tries=5, delay=5, backoff=2.0)
+def list_iam_server_certificates_with_backoff(connection, path):
+    paginator = connection.get_paginator('list_server_certificates')
+    if path:
+        return paginator.paginate(PathPrefix=path).build_full_result()
+    else:
```

Comment:
Since you always return in your if statement, this else isn't technically required.

```
if path:
  return ....

return ....
```